### ISSUE TITLE
`syntaxError` in include is hidden leading to suspected false negative

### ISSUE THREAD

**Issue by firewave** [johnthagen/clion-cppcheck#74]
`test.h`
```cpp
enum Type
{
	ERROR
};

struct A
{
};
```

`test.cpp`
```cpp
#include "test.h"

extern void f()
{
    A a; // unusedVariable
}
```

In Cppcheck 2.4 the header will generate a `syntaxError` with the Windows version:
```
test.h:1:1: error: syntax error [syntaxError]
enum Type
^
```
This leads to the `unusedVariable` in function `f()` not being reported making it seems like a false negative since the `syntaxError` is being suppressed as it didn't occur in the file we are analyzing.

We need to make this error visible - possibly via a global warning about this.

There's most likely other errors which may lead to the same problem.

**Comment by johnthagen** [comment]
I wonder if there is a hook to simply not run if there are syntax errors. I know that is how many people set up File Watchers in IntelliJ products.

**Comment by firewave** [comment]
A preview of the fix I am currently testing:

![image](https://user-images.githubusercontent.com/242857/158177969-f88bbb7a-2190-4c9d-8dc8-1a0da4d4e1d4.png)

I am not sure if you can put links in such in messages. Being able to directly jump to the file/line in question would be awesome.

**Comment by firewave** [comment]
Note: A potential fix for https://trac.cppcheck.net/ticket/11913 might affect this.
